recipes/install.rb

```diff
@@ -393,17 +393,38 @@
     user "root"
     code <<-EOH
     set -e
-    if [ ! -d "/cgroup" ] ; then
-       mkdir /cgroup
+    if [ ! -d "/sys/fs/cgroup/cpu/hops-yarn" ] ; then
+       mkdir /sys/fs/cgroup/cpu/hops-yarn
     fi
-    mount -t cgroup -o cpu cpu /cgroup
+    if [ ! -d "/sys/fs/cgroup/devices/hops-yarn" ] ; then
+       mkdir /sys/fs/cgroup/devices/hops-yarn
+    fi
+    # mount -t cgroup -o cpu cpu /cgroup
     touch #{cgroups_mounted}
   EOH
      not_if { ::File.exist?("#{cgroups_mounted}") }
   end
 
 end
 
+bash 'update_owner_for_gpu' do
+  user "root"
+  code <<-EOH
+    set -e
+    chown root:#{node.hops.group} #{node.hops.dir}
```

Comment:
For below, it is better to have "chmod 750" or whatever the permissions are. When you call chmod go-w, you need to know what the previous permissions were to know what they will be afterwards. Harder to understand the code.